src/refactor_nrepl/analyzer.clj

```diff
@@ -91,7 +91,12 @@
     (when ns
       (if-let [cached-ast-or-err (get-ast-from-cache ns file-content)]
         cached-ast-or-err
-        (when-let [new-ast-or-err (try (build-ast ns aliases) (catch Throwable th th))]
+        (when-let [new-ast-or-err (try
+                                    (build-ast ns aliases)
+                                    (catch Throwable th
+                                      (when (System/getProperty "refactor-nrepl.internal.log-exceptions")
```

Comment:
Might be nice to pull this into a helper.  Not would would that read better but would make it easier to re-use without looking up the name of the property.

When doing 'information hiding' like this, I like to use the prefix `maybe-` for conditional actions like this.  `maybe-debug-log-exception`?  That makes code like this very easy to skip (if you don't care about the encapsulated details) during code reading later.